server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java

```diff
@@ -118,7 +118,7 @@ public void writeFile(StoreFileMetadata fileMetadata, long readSnapshotFileBuffe
             }
 
             Store.verify(indexOutput);
-            assert Arrays.asList(store.directory().listAll()).contains(tempFileName)
+            assert verifyOutput == false || Arrays.asList(store.directory().listAll()).contains(tempFileName)
```

Comment:
:+1:

I suspect we need something similar in `innerWriteFileChunk` too?

Reply:
I think we're not writing chunks for where we're looking to use this are we?

Reply:
I double checked and you're right, it should not be needed